[Dogfood] ESLint config should exempt test files from complexity limits

**Title:** [Dogfood] ESLint config should exempt test files from complexity limits

**Found During:** Dogfooding test (Phase 6)

**Severity:** 🟡 High - AGENTS.md claims exemption exists but config doesn't implement it

---

## Description

During dogfood testing, discovered that test files are penalized with complexity violations even though **AGENTS.md explicitly states**: "Test files exempt from line/complexity limits" (line 114).

The generated `eslint.config.js` does NOT implement this exemption, causing test files to fail for being "too long" when they're legitimately testing complex scenarios.

---

## Current Behavior

**Test violations found:**
```
tests/integration/cli-learn-interactive-snapshot.test.js:
  65:48  warning  Function has too many lines (156). Maximum allowed is 100
  95:54  warning  Function has too many lines (133). Maximum allowed is 100

tests/integration/cli-init-interactive.test.js:
  8:42  warning  Function has too many lines (158). Maximum allowed is 100
  9:41  warning  Function has too many lines (117). Maximum allowed is 100
```

**Generated eslint.config.js:**
```javascript
export default [
  {
    files: ['**/*.js'],
    rules: {
      'max-lines-per-function': ['warn', 100],  // ❌ Applies to ALL files
      'complexity': ['warn', 15],                // ❌ Applies to ALL files
      'max-depth': ['warn', 4],                  // ❌ Applies to ALL files
    }
  }
];
```

**No exemption override for test files!**

---

## Expected Behavior

Generated `eslint.config.js` should include override for test files:

```javascript
export default [
  {
    files: ['**/*.js'],
    rules: {
      'max-lines-per-function': ['warn', 100],
      'complexity': ['warn', 15],
      'max-depth': ['warn', 4],
    }
  },
  {
    // ✅ Exemptions for test files
    files: [
      '**/*.test.js',
      '**/*.spec.js', 
      'tests/**/*.js',
      'test/**/*.js'
    ],
    rules: {
      'max-lines-per-function': 'off',  // ✅ Tests can be long
      'complexity': 'off',               // ✅ Tests can be complex
      'max-statements': 'off',           // ✅ Tests need many assertions
      'max-depth': 'off'                 // ✅ Tests can be deeply nested
    }
  }
];
```

---

## Root Cause

**File:** `lib/generators/eslint-config.js`

The ESLint config generator:
1. Creates rules for all files
2. **Does NOT add test file override block**

**AGENTS.md claims (line 114):**
```markdown
**Note**: Test files exempt from line/complexity limits
```

But the generator doesn't implement this promise.

---

## Impact

**Users:**
- Test files flagged with false violations
- Legitimate long test functions penalized
- Confusing warnings in test runs
- AGENTS.md is misleading

**Example:** Integration tests that mock user interaction workflows are inherently long:
```javascript
// tests/integration/cli-learn-interactive-snapshot.test.js
test('learn interactive workflow with prompts', async () => {
  // Setup: 30 lines
  // Mock prompts: 40 lines  
  // Execute command: 20 lines
  // Verify results: 30 lines
  // Cleanup: 20 lines
  // Total: 140 lines ❌ Violation!
});
```

This is a **legitimate test** but gets flagged.

---

## Proposed Fix

### Update `lib/generators/eslint-config.js`

Add test file overrides to generated config:

```javascript
function writeEslintConfig(cwd, cfg) {
  // ... existing code ...
  
  const config = [
    js.configs.recommended,
    {
      files: ['**/*.js'],
      // ... existing rules ...
    },
    // ✅ ADD THIS BLOCK
    {
      files: [
        '**/*.test.js',
        '**/*.spec.js',
        'tests/**/*.js',
        'test/**/*.js'
      ],
      rules: {
        'max-lines-per-function': 'off',
        'complexity': 'off',
        'max-statements': 'off',
        'max-depth': 'off',
        'max-nested-callbacks': 'off'
      }
    }
  ];
  
  // ... write config ...
}
```

---

## Files Affected

- `lib/generators/eslint-config.js` (add test override block)
- `AGENTS.md` (already documents this correctly)

---

## Testing

After fix, regenerate config and verify:

```bash
# Clean and regenerate
rm eslint.config.js
node bin/cli.js init --primary=dev-tools --yes --agents

# Run ESLint on tests - should have NO complexity violations
npx eslint tests/ 2>&1 | grep -E "max-lines-per-function|complexity|max-statements"
# Expected: no output (tests exempt)

# Verify override exists in config
grep -A 10 "test.*js" eslint.config.js
# Expected: see override block with 'off' rules
```

---

## Context

**Test Branch:** `test/dogfood-validation`

**Related:**
- Issue #107 (complexity in source files - separate issue)
- AGENTS.md line 114 (documents exemption)
- DOGFOOD_TEST_PLAN.md line 505 (expects exemption)

**Dogfood results:** See `dogfood-results.md` Phase 6

---

## Acceptance Criteria

- [ ] Generated `eslint.config.js` includes test file override block
- [ ] Test files exempt from `max-lines-per-function`
- [ ] Test files exempt from `complexity`
- [ ] Test files exempt from `max-statements`
- [ ] Test files exempt from `max-depth`
- [ ] Exemption applies to `*.test.js`, `*.spec.js`, and `tests/**/*.js`
- [ ] Running ESLint on `tests/` shows no complexity violations
- [ ] Source files still have complexity rules applied
- [ ] All existing tests pass

---

## Priority

**High** - This is a **dogfooding failure**. Our documentation claims behavior that doesn't exist, and our own test suite gets penalized unfairly.

--- a/lib/generators/eslint-arch-config.js
+++ b/lib/generators/eslint-arch-config.js
@@ -81,7 +81,8 @@ function generateArchitectureRules(architecture) {
     rules: {
       'max-lines-per-function': 'off',
       'max-statements': 'off',
-      'complexity': 'off'
+      'complexity': 'off',
+      'max-depth': 'off'
     }
   });
 

--- a/tests/lib/generators/eslint-arch-config.test.js
+++ b/tests/lib/generators/eslint-arch-config.test.js
@@ -121,6 +121,7 @@ describe('eslint-arch-config generator', function () {
       assert.strictEqual(testOverride.rules['max-lines-per-function'], 'off');
       assert.strictEqual(testOverride.rules['max-statements'], 'off');
       assert.strictEqual(testOverride.rules['complexity'], 'off');
+      assert.strictEqual(testOverride.rules['max-depth'], 'off');
     });
   });
 

